lte/gateway/python/magma/pipelined/app/mandatory_service.py

```diff
@@ -0,0 +1,179 @@
+"""
+Copyright 2022 The Magma Authors.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+This file models common logic of the mandatory services ingress,
+middle and egress.
+"""
+from magma.pipelined.app.base import MagmaController
+from magma.pipelined.app.restart_mixin import DefaultMsgsMap, RestartMixin
+from magma.pipelined.openflow import flows
+from magma.pipelined.openflow.magma_match import MagmaMatch
+from magma.pipelined.openflow.messages import MessageHub, MsgChannel
+from magma.pipelined.openflow.registers import TUN_PORT_REG, Direction
+from ryu.controller import ofp_event
+from ryu.controller.handler import MAIN_DISPATCHER, set_ev_cls
+
+
+class MandatoryController(RestartMixin, MagmaController):
```

Comment:
Hm I'm not sure why we're adding unnecessary inheritance here, whats the purpose of MandatoryController class?

Reply:
I think we shouldn't introduce this

Reply:
The reason is the common logic used in the three new controllers - would you prefer to handle this in a util class? (at least the ryu annotated functions would need to be duplicated in ingress, middle and egress apps).

Reply:
Yeah we already have a MagmaController class, lets either put stuff there or duplicate the code

Reply:
Hi Nick, MagmaController is super class of a lot of apps so we should not move the common logic for ingress, middle and egress there. We will remove the new abstract controller and inline the code.

> f.e this seems to be only used in middle
_get_vlan_egress_flow_msgs is used in middle and egress and please note that this does not belong to the class MandatoryController - this is the same patter as in inout.py before. We will move it to a util file/module.

Reply:
pushed - running integ test on fork to be sure